tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/packaging/ModuleExecutionFlowTests.java

```diff
@@ -212,26 +212,39 @@ public void testStopHandlerExecutionOrder() throws BallerinaTestException {
     }
 
     @Test
-    public void testListenerStopHandlerAsyncCall() throws BallerinaTestException {
+    public void testListenerStopHandlerShutdownOrderWithErrorReturn() throws BallerinaTestException {
         Path projectPath = Paths.get("src", "test", "resources", "packaging",
-                "listener_stophandler_async_call_test");
+                "listener_stophandler_shutdown_order_project");
 
         BServerInstance serverInstance = new BServerInstance(balServer);
         serverInstance.startServer(projectPath.toAbsolutePath().toString(), projectPath.getFileName().toString(), null,
                 null, null);
-        LogLeecher logLeecherD = new LogLeecher("StopHandler2 of current module");
-        LogLeecher logLeecherE = new LogLeecher("StopHandler1 of current module");
-        LogLeecher logLeecherF = new LogLeecher("StopHandler2 of moduleA");
-        LogLeecher logLeecherG = new LogLeecher("StopHandler1 of moduleA");
-        LogLeecher logLeecherH = new LogLeecher("graceful stop of current module", LogLeecher.LeecherType.ERROR);
-        LogLeecher logLeecherI = new LogLeecher("graceful stop of ModuleA", LogLeecher.LeecherType.ERROR);
-        serverInstance.addLogLeecher(logLeecherD);
+        LogLeecher logLeecherA = new LogLeecher("Calling stop for static listener");
```

Comment:
Shall we add tests for panic cases as well? in `stop-handler`, `static listeners` , and `dynamic listeners`?

Reply:
If there a panic or error return inside any call of gracefulStop() or stopHandler(), then the program terminates unsuccessfully by panicking or returning the error value. Already a test case available to cover this case https://github.com/ballerina-platform/ballerina-lang/blob/c7c5ee4745f702532c6f24821e2e9fa093de676a/tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/packaging/ModuleExecutionFlowTests.java#L105